apps/next/src/pages/Portfolio/components/PortfolioHome/components/PortolioDetails/components/FilterMenu.tsx

```diff
@@ -0,0 +1,85 @@
+import {
+  List,
+  ListItemButton,
+  ListItemText,
+  Popover,
+} from '@avalabs/k2-alpine';
+import { FC } from 'react';
+import { useTranslation } from 'react-i18next';
+import { MdCheck } from 'react-icons/md';
+import { NetworkWithCaipId } from '@core/types';
+
+export type FilterMenuProps = {
+  id: string;
+  anchorEl: HTMLButtonElement | null;
+  open: boolean;
+  onClose: () => void;
+  selectedNetworks: Set<number>;
+  setSelectedNetworks: (networks: Set<number>) => void;
+  availableNetworks: NetworkWithCaipId[];
+};
```

Comment:
Please consider naming props by their function rather than derive it from the consumer's state names.
Also, if the selected and onChange props are an underlying property of the data item, it is useful to specify which exact property is expected (here: `number` => `NetworkWithCaipId['chainId']`).

```suggestion
export type FilterMenuProps = {
  id: string;
  anchorEl: HTMLButtonElement | null;
  open: boolean;
  onClose: () => void;
  selected: Set<NetworkWithCaipId['chainId']>;
  onChange: (networks: Set<NetworkWithCaipId['chainId']>) => void;
  networks: NetworkWithCaipId[];
};
```